tfrecorder/input_schema.py

```diff
@@ -0,0 +1,163 @@
+# Lint as: python3
+
+# Copyright 2020 Google LLC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Defines input types for TFRecorder's input schema."""
+
+from collections import OrderedDict
+from dataclasses import dataclass
+from typing import Any, Dict, List
+
+import frozendict
+import tensorflow as tf
+import tensorflow_transform as tft
+from tensorflow_transform.tf_metadata import dataset_metadata
+from tensorflow_transform.tf_metadata import schema_utils
+
+from tfrecorder import types
+
+@dataclass
+class SupportedType:
+  """Base type for TFRecorder Types."""
+  feature_spec: tf.io.FixedLenFeature
+  allowed_values: List[Any]
+
+@dataclass
+class ImageUriType(SupportedType):
+  """Supports image uri columns."""
+  feature_spec=tf.io.FixedLenFeature([], tf.string)
+  allowed_values=[]
+
+@dataclass
+class SplitKeyType(SupportedType):
+  """Supports split key columns."""
+  feature_spec=tf.io.FixedLenFeature([], tf.string)
+  allowed_values = ['TRAIN', 'VALIDATION', 'TEST', 'DISCARD']
+
+@dataclass
+class IntegerInputType(SupportedType):
+  """Supports integer columns."""
+  feature_spec=tf.io.FixedLenFeature([], tf.int64)
+  allowed_values=[]
+
+@dataclass
+class FloatInputType (SupportedType):
+  """Supports float columns."""
+  feature_spec=tf.io.FixedLenFeature([], tf.float64)
+  allowed_values=[]
+
+#TODO(mikebernico): Implement in preprocess_fn
+@dataclass
+class StringInputType(SupportedType):
+  """Supports string input columns."""
+  feature_spec=tf.io.FixedLenFeature([], tf.string)
+  allowed_values=[]
+
+@dataclass
+class IntegerLabelType(SupportedType):
+  """Supports integer labels."""
+  feature_spec=tf.io.FixedLenFeature([], tf.int64)
+  allowed_values=[]
+
+@dataclass
+class StringLabelType(SupportedType):
+  """Supports string labels."""
+  feature_spec=tf.io.FixedLenFeature([], tf.string)
+  allowed_values=[]
+
+@dataclass
+class ImageSupportStringType(SupportedType):
+  """Supports generated image bytestrings."""
+  feature_spec=tf.io.FixedLenFeature([], tf.string)
+  allowed_values=[]
+
+@dataclass
+class ImageSupportIntType(SupportedType):
+  """Supports generated image ints (height, width, channels)."""
+  feature_spec=tf.io.FixedLenFeature([], tf.int64)
+  allowed_values=[]
+
+class Schema:
+  """Defines a TFRecorder input schema."""
+  def __init__(self,
+               schema_map: OrderedDict
```

Comment:
Maybe this can just take in a regular `dict` and have the OrderedDict constructed in the attribute.

Reply:
Done